### PR TITLE
Fixed react-hooks dependency lists errors and warnings

### PR DESCRIPTION
**Pull request by jeremyphilemon**
- Thanks to @ssonal!

Also there seemed to be a problem with the minigraphs rendering. Having `timeseries`, `Graphdata`, and `time series.length` in the dependency list redrew the d3 graphs multiple times. I've disabled that for now so if you can come up with a fix for it next, we can start enforcing the react-hooks rules in the eslint.js files :)

### CODE REVIEW

File: src/components/minigraph.js [hunk 1]
```diff
@@ -29,274 +23,283 @@ function Minigraph(props) {
...
-      })
-      .style('opacity', 0)
-      .transition()
-      .duration(props.animate ? 3000 : 0)
-      .style('opacity', 1);
-  }
+  useEffect(() => {
+    graphData(timeseries);
+    // eslint-disable-next-line
+  }, [timeseries]);
```
Comment (jeremyphilemon) [inline src/components/minigraph.js:302]: @ssonal here.

Comment (ssonal) [reply src/components/minigraph.js:302]: Just patched this in #402. Should work now!

Side-note, testing these conditions is pretty painful. How are you doing it?

Comment (jeremyphilemon) [reply src/components/minigraph.js:302]: Haha I feel you, it's a pain! 

So I noticed the graphs looked odd, so inspected the code and added a `console.log` in the useEffect to see how many times the state changes happened in the dependency list, and voila! The graphData was called so many times having it redrawn so many times.

Could've slipped my eyes, but hey :P

Comment (jeremyphilemon) [reply src/components/minigraph.js:302]: @ssonal also I had a look at the time series graphs, I think it's the same issue. The graphs tends to be redrawn again. 

See if you can fix it too!

**Pull request merged**